src/main/java/uk/gov/ons/ctp/integration/rhsvc/event/impl/EventFilter.java

```diff
@@ -0,0 +1,71 @@
+package uk.gov.ons.ctp.integration.rhsvc.event.impl;
+
+import static uk.gov.ons.ctp.common.log.ScopedStructuredArguments.kv;
+
+import java.util.Optional;
+import java.util.Set;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.ctp.common.error.CTPException;
+import uk.gov.ons.ctp.common.event.model.CollectionExercise;
+import uk.gov.ons.ctp.common.event.model.SurveyUpdate;
+import uk.gov.ons.ctp.integration.rhsvc.config.AppConfig;
+import uk.gov.ons.ctp.integration.rhsvc.repository.RespondentDataRepository;
+
+@Slf4j
+@Component
+public class EventFilter {
+
+  private AppConfig appConfig;
+
+  private RespondentDataRepository respondentDataRepo;
+
+  public EventFilter(AppConfig appConfig, RespondentDataRepository respondentDataRepo) {
+    this.appConfig = appConfig;
+    this.respondentDataRepo = respondentDataRepo;
+  }
+
+  public boolean isValidEvent(String surveyId, String collexId, String caseId, String messageId)
+      throws CTPException {
+
+    Optional<SurveyUpdate> surveyUpdateOpt = respondentDataRepo.readSurvey(surveyId);
+    if (surveyUpdateOpt.isPresent()) {
+      SurveyUpdate surveyUpdate = surveyUpdateOpt.get();
+      if (isAcceptedSurveyType(surveyUpdate.getSampleDefinitionUrl())) {
+        Optional<CollectionExercise> collectionExercise =
+            respondentDataRepo.readCollectionExercise(collexId);
+        if (collectionExercise.isPresent()) {
+          return true;
+        } else {
+          // TODO - should we NAK the event/throw exception if we do not recognize the collex and
```

Comment:
Im out of the loop of this project, do these TODOs need to be captured?
Or will they natrually come out of the wash.